mkosi/__init__.py

```diff
@@ -4458,6 +4460,70 @@ def extract_unified_kernel(
     return f
 
 
+def extract_kernel_image_initrd(
+    args: MkosiArgs,
+    root: Path,
+    do_run_build_script: bool,
+    for_cache: bool,
+    mount: Callable[[], ContextManager[None]],
+) -> Tuple[Optional[BinaryIO], Optional[BinaryIO]]:
```

Comment:
Could this maybe be made more precise like this?
```suggestion
) -> Union[Tuple[BinaryIO], BinaryIO], Tuple[None, None]]:
```

This looks more ugly now, but in the future (3.10+) this will would actually be nicely readable
```python
tuple[BinaryIO, BinaryIO] | tuple[None, None]
```
(while also possibly being more precise)